lib/src/components/NxTextInput/types.ts

```diff
@@ -10,10 +10,12 @@ import { FormEvent, InputHTMLAttributes, TextareaHTMLAttributes, ReactNode } fro
 /**
  * The valid values for the `type` Prop
  */
-export const inputTypes = ['textarea', 'text', 'password'] as const;
+export const publicInputTypes = ['textarea', 'text', 'password'] as const;
+export const inputTypes = [...publicInputTypes, 'date'] as const;
 
 // See https://stackoverflow.com/a/45486495
-export type NxTextInputType = (typeof inputTypes)[number];
+type NxTextInputType = (typeof inputTypes)[number];
+export type PublicNxTextInputType = (typeof inputTypes)[number];
```

Comment:
Ah this is the reason the issue I pointed out above wasn't caught by the compiler.
```suggestion
export type PublicNxTextInputType = (typeof publicInputTypes)[number];
```